### ISSUE TITLE
usesCleartextTraffic needed also with custom headers

### ISSUE THREAD

**Issue by enricobenedos** [ryanheise/just_audio#420]
**To which pages does your suggestion apply?**

- https://github.com/ryanheise/just_audio/blob/master/just_audio/README.md

**Quote the sentences(s) from the documentation to be improved (if any)**

> If you wish to connect to non-HTTPS URLS, also add the following attribute to the application element:
>`<application ... android:usesCleartextTraffic="true">`

**Describe your suggestion**
Good morning, I lost a lot of time understanding why this plugin was not working on Android but it was working on iOS. This is my code:
```dart
await _player.setUrl(widget.trackUrl, headers: HTTPService.getHeaders());
```
on documentation it is suggested to set `usesCleartextTraffic` in case you need to use not HTTPS urls. But in my case I'm using HTTPS urls but on the documentation it is not reported that:
> ... If headers are set, just_audio will create a cleartext local HTTP proxy on your device to forward HTTP requests with headers included.

I suggest to change the documentation from:

> If you wish to connect to non-HTTPS URLS, also add the following attribute to the application element:
>`<application ... android:usesCleartextTraffic="true">`

to

> If you wish to connect to non-HTTPS URLS or use custom headers, also add the following attribute to the application element:
>`<application ... android:usesCleartextTraffic="true">`

Thank you

**Comment by ryanheise** [comment]
I agree this could be beneficial to add to the readme. It is possibly mentioned or implied in the API documentation though if I'm not mistaken.

I would eventually like to implement headers without a proxy which would also solve this but I would welcome a pull request if anyone wishes to help
